File: frontend/labelix-web-app/src/app/utility/contracts/IProject.ts
import {IIdentifiable} from './IIdentifiable';
import {IData} from './IData';
import {ICocoFormat} from './cocoFormat/ICocoFormat';

export interface IProject extends IIdentifiable{
  name: string;
  description: string;
  creationDate: Date;
  // creator: string; // later here will be another interface
  finishedAnnotation: boolean;
  label: string;
  images: IData[];
  timestamp: Date;
  AIModelConfig: number[];
  cocoExport: ICocoFormat;
}

File: backend/Labelix-Backend/Labelix.Contracts/Persistence/IProject.cs
using System;
using System.Collections.Generic;

namespace Labelix.Contracts.Persistence
{
    public interface IProject : IIdentifiable, ICopyable<IProject>
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreationDate { get; set; }
        public bool FinishedAnnotation { get; set; }
        public string LabeledPath { get; set; }
    }
}
